src/test/ui/mismatched_types/abridged.rs

```diff
@@ -50,4 +50,8 @@ fn e() -> X<X<String, String>, String> {
     x //~ ERROR mismatched types
 }
 
+fn f() -> String {
+    1+2 //~ ERROR mismatched types
```

Comment:
The same issue exists with unary operators, eg. `-2` will suggest `-2.to_string()`, which doesn't compile. Might make sense to add another case for `ExprKind::Unary` above.

Reply:
Sure